src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -97,21 +97,28 @@ private TypeSyntax CreateTypeSyntaxForSymbol(ITypeSymbol typeSymbol)
         }
         // TODO optimize
         else if (typeSymbol is INamedTypeSymbol namedTypeSymbol &&
-            namedTypeSymbol.TypeArguments.Length == 1 &&
+            namedTypeSymbol.TypeArguments.Length >= 1 &&
             namedTypeSymbol.IsSupportedCollection())
         {
-            var underlyingTypeSyntax = CreateTypeSyntaxForSymbol(namedTypeSymbol.TypeArguments.Single());
-            var listSyntax = SyntaxFactory.GenericName(
-                SyntaxFactory.Identifier("List"),
-                SyntaxFactory.TypeArgumentList(SyntaxFactory.SeparatedList(new[] { underlyingTypeSyntax })));
+            var underlyingTypeSyntaxes = namedTypeSymbol.TypeArguments.Select(typeArgument => CreateTypeSyntaxForSymbol(typeArgument));
+            var collectionIdentifier = namedTypeSymbol.TypeArguments.Length switch
```

Comment:
We need a better collection identification here.
What if it's `MyCustomList<A,B> : IList<A>`  (probably good idea to add tests).

Let's determine the collection type by the actual type: Dictionary, IDictionary => Dictionary, List ... => List